### PR TITLE
DOC: Update RTD settings

### PR DESCRIPTION
**Pull request by pllim**
### Description

Update RTD settings because they are removing Sphinx context injection at build time. Got a memo in email about this. Also see https://github.com/readthedocs/readthedocs.org/issues/11474 and https://about.readthedocs.com/blog/2024/07/addons-by-default/

Preview at https://astropy--16737.org.readthedocs.build/en/16737/ though it is hard to see how things will really be when RTD remove things on their end.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -270,6 +273,8 @@
     "github_repo": "astropy",
     "github_version": "main",
     "doc_path": "docs",
+    # Tell Jinja2 templates the build is running on Read the Docs
+    "READTHEDOCS": os.environ.get("READTHEDOCS", "") == "True",
```
Comment (pllim) [inline docs/conf.py:277]: @humitos , can I do it like this, or does this have to be nested in an `if` strictly as stated in the email I received?

```python
import os

# Set canonical URL from the Read the Docs Domain
html_baseurl = os.environ.get("READTHEDOCS_CANONICAL_URL", "")

# Tell Jinja2 templates the build is running on Read the Docs
if os.environ.get("READTHEDOCS", "") == "True":
    html_context["READTHEDOCS"] = True
```

**Comment by humitos** [comment]
> Preview at [astropy--16737.org.readthedocs.build/en/16737](https://astropy--16737.org.readthedocs.build/en/16737/) though it is hard to see how things will really be when RTD remove things on their end.

I can enable the `DISABLE_SPHINX_MANIPUALTION` feature flag in your project if you want to perform the tests. That flag, together with [enabling Read the Docs Addons manually](https://docs.readthedocs.io/page/addons.html#enabling-read-the-docs-addons) emulates the default future behavior. Let me know.

File: docs/conf.py (hunk 1 above)
Comment (humitos) [reply docs/conf.py:277]: Yes, you can do it like you've done it here. The snipped in the email is just an example 👍🏼

**Comment by pllim** [comment]
> It's worth to note that this feature flag will disable the installation of readthedocs-sphinx-ext and won't append any config to the conf.py (this are the behaviors we are deprecating)

So, will the "flyout" still appear for a previously built released version of doc?

**Comment by humitos** [comment]
> So, will the "flyout" still appear for a previously built released version of doc?

Yes. It will use the new flyout addons, tho, since that feature is globally applied to all the versions of the project when enabled.

**Comment by pllim** [comment]
Thanks for the clarification, @humitos ! How does this plan sound to you then?

1. You enable the `DISABLE_SPHINX_MANIPULATION` thingy.
2. I merge this PR.
3. I look at the docs (latest and stable).
4. If I do not like them, you can undo (1). Otherwise, nothing else to do.

**Comment by humitos** [comment]
Sounds great to me! 👍🏼 

I can enable it _right now_ and stay for some time here in case you want me to disable it. Let me know.

**Comment by humitos** [comment]
Should I enable it now?

**Comment by pllim** [comment]
Yes, please!

**Comment by humitos** [comment]
Done. I enabled `DISABLE_SPHINX_MANIPULATION` in `astropy` project. Let me know if everything works as you expected. I'd really appreciate your feedback 👍🏼

**Comment by pllim** [comment]
I think it is fine but I'll be on the lookout for others reporting problems. Thanks again!

**Comment by humitos** [comment]
Excellent! Don't hesitate to ping me if you have any questions 👍🏼 I hope everything keeps working fine and the transition to be as smooth as possible.

**Pull request merged**